model/gen-ai/registry.yaml

```diff
@@ -13,9 +13,9 @@ groups:
               stability: experimental
               value: "openai"
               brief: 'OpenAI'
-            - id: vertex_ai
+            - id: vertex.ai
               stability: experimental
-              value: "vertex_ai"
+              value: "vertex.ai"
```

Comment:
While we're here, should we prefix with `gcp.` like we have for some other vendors?

Reply:
we're really inconsistent on prefixing things with cloud name (not intentionally).
E.g. we have [`spanner`](https://github.com/open-telemetry/semantic-conventions/blob/0d843e4cb842c816559072d89ede1fd5544f1657/docs/attributes-registry/db.md?plain=1#L149) and it's ok since there is just one spanner DB.

We have [`gcp_pubsub`](https://github.com/open-telemetry/semantic-conventions/blob/0d843e4cb842c816559072d89ede1fd5544f1657/docs/messaging/gcp-pubsub.md) for messaging because `pubsub` is too generic..

We also have some `gcp` [attributes](https://github.com/open-telemetry/semantic-conventions/blob/0d843e4cb842c816559072d89ede1fd5544f1657/docs/attributes-registry/gcp.md) and [resources](https://github.com/open-telemetry/semantic-conventions/tree/0d843e4cb842c816559072d89ede1fd5544f1657/docs/resource/cloud-provider/gcp)

Both `vertex.ai` or `vertext_ai` are unique enough - they don't **have to** have a prefix similarly to `spanner`. If it does, it should be `gcp`.

So what should we do:
1. `spanner`, `vertex_ai`, `gcp_pubsub`
2. `spanner`, `vertex_ai`, `gcp.pubsub`
3. `gcp.spanner`, `gcp.vertex_ai`, `gcp.pubsub`
4. `gcp_spanner`, `gcp_vertex_ai`, `gcp_pubsub`
5. ...
?

Speaking from Azure side, we're likely to stay where we are: `cosmosdb`, `eventhubs`, `az.ai.inference` (use product name when it's unique enough, use `az.` namespace when it's not). Reasons: it's nice to be consistent, but some of these things are de-facto stable and are used wide enough to make breaking really problematic.

General issue - https://github.com/open-telemetry/semantic-conventions/issues/608

cc @jsuereth if he has any thoughts

Reply:
There's also the [`cloud.platform` values](https://github.com/open-telemetry/semantic-conventions/blob/0d843e4cb842c816559072d89ede1fd5544f1657/docs/attributes-registry/cloud.md#:~:text=cloud.platform%20has%20the%20following%20list%20of%20well%2Dknown%20values.%20If%20one%20of%20them%20applies%2C%20then%20the%20respective%20value%20MUST%20be%20used%3B%20otherwise%2C%20a%20custom%20value%20MAY%20be%20used.) which look like `azure_functions`, `gcp_cloud_functions `.

Reply:
I'm like 80% confident we can rename `azure_functions` to `az.functions`. Let me try